scripts/install-du.ps1

```diff
@@ -94,30 +97,71 @@ function CalcHash {
     [Convert]::ToBase64String($hashAlg.ComputeHash([System.IO.File]::ReadAllBytes($FilePath)))
 }
 
-function WriteExtensionRegistration {
+function WriteContentDownloaderRegistration{
     Param(
-        [Parameter(Mandatory = $true)][string]$BinaryFilePath,
-        [Parameter(Mandatory = $true)][string]$OutputJsonFilePath
+        [Parameter(Mandatory = $true)][string]$BinaryFileNameNoExt,
+        [Parameter(Mandatory = $true)][string]$ExtensionsDir
     )
 
-    $fileSizeInBytes = (Get-Item $BinaryFilePath).Length
-    $fileHash = CalcHash -FilePath $BinaryFilePath
+    $adu_extensions_sources_dir = "$ExtensionsDir/sources"
+    $binaryFilePath = "$adu_extensions_sources_dir/${BinaryFileNameNoExt}.dll"
+
+    $fileSizeInBytes = (Get-Item $binaryFilePath).Length
+    $fileHash = CalcHash -FilePath $binaryFilePath
 
     $registration = [ordered]@{
-        fileName="$BinaryFilePath"
-        sizeInBytes=$FileSizeInBytes
+        fileName="$binaryFilePath"
+        sizeInBytes=$fileSizeInBytes
         hashes = @{
             sha256="$fileHash"
         }
     }
 
-    $registration | ConvertTo-Json | Set-Content $OutputJsonFilePath
+    $registration | ConvertTo-Json | Set-Content "$ExtensionsDir/content_downloader/extensions.json"
+}
+
+function WriteContentHandlerRegistration{
```

Comment:
I originally didn't include this as I'm concerned about the format of the registration file changing and this breaking.  

Maybe a better approach would be rather than calling into the handler to do the registration as we do now, the handler has something like "--GetRegistrationJson --InstallationPath foo" which would output the json so that this script and the agent could use that to write the file.